pkg/sentry/fs/fsutil/cache_reclaim.go

```diff
@@ -0,0 +1,350 @@
+package fsutil
+
+import (
+        "sync"
+        "sync/atomic"
+        "time"
+        "runtime"
+
+        "gvisor.dev/gvisor/pkg/ilist"
+        "gvisor.dev/gvisor/pkg/log"
+        "gvisor.dev/gvisor/pkg/sentry/context"
+        "gvisor.dev/gvisor/pkg/sentry/memmap"
+        "gvisor.dev/gvisor/pkg/sentry/pgalloc"
+        "gvisor.dev/gvisor/pkg/sentry/platform"
+        "gvisor.dev/gvisor/pkg/sentry/usermem"
+)
+
+const (
+        LRU_ACTIVE_LIST = 0
+        LRU_INACTIVE_LIST = 1
+        LRU_LIST_NUM = 2
+)
+
+type LruEntry struct {
+	ilist.Entry
```

Comment:
You can use the templated version of ilist to make this more efficient. (Otherwise, all the operations for storing the LruEntries require boxing and unboxing.)

Reply:
OK, I'll do it. But what difference does template in gvisor have, I'm not ver experienced in go.

Reply:
The template just means that everything is properly typed. It will have a minor performance impact here, but it's mostly about consistency.